packages/workspace/src/core/file-utils.ts

```diff
@@ -35,29 +44,30 @@ export function calculateFileChanges(
     const ext = extname(f);
     const _mtime = mtime(`${appRootPath}/${f}`);
     // Memoize results so we don't recalculate on successive invocation.
-    let value: any[] = null;
 
     return {
       file: f,
       ext,
       mtime: _mtime,
-      getChanges: () => {
-        if (!value) {
-          switch (ext) {
-            case '.json':
-              if (base) {
-                const atBase = readFileAtRevision(f, base);
-                const atHead = readFileAtRevision(f, head);
-                value = jsonDiff(atBase, atHead);
-              } else {
-                value = [];
-              }
-              break;
-            default:
-              throw new Error(`Cannot call getChanges() on ${f}`);
-          }
+      getChanges: (): Modification[] => {
+        if (!nxArgs) {
+          return [WholeFileChange];
+        }
+        switch (ext) {
+          case '.json':
+            const atBase = nxArgs.base
+              ? readFileAtRevision(f, nxArgs.base)
+              : readFileAtRevision(f, 'master');
```

Comment:
should not hardcode master here. We should default base to master.

Reply:
We have to refactor the flow of processing the args a bit to do this. I'd rather do that separately.

Reply:
ok